fasjson/web/app.py

```
@@ -27,6 +31,26 @@ class NameConverter(BaseConverter):
 app.url_map.converters["name"] = NameConverter
 
 
+def ldap_running():
+    """Readiness Health Check"""
+    try:
+        client = ldap.initialize(app.config["FASJSON_LDAP_URI"])
+        client.simple_bind_s()
+        return True, "OK"
+    except ldap.SERVER_DOWN:
+        return False, "NOT OK"
+
+
+def basic_liveness():
+    """Liveness Health Check"""
+    return True, "OK"
+
+
+health_check = HealthCheck(app)
```

The extension approach is interesting but I was more thinking of a [blueprint](https://flask.palletsprojects.com/en/1.1.x/blueprints/) approach, this way it will be easier to declare the endpoints and the user only has to set the mount point (here, `healthz`).

Also, I think that the actual methods could be read from the app's configuration file, that seems easier to use for apps that will use this library, but that's debatable.
I would also recommend defining an exception that the user could raise to indicate that something is wrong, rather than returning a tuple.

---

OK yeah fair point. I've swapped to blueprints now. Its also configured using an app config key, if thats not provided then you can still add the check manually using `add_check` before registering the blueprint if you want. Let me know what you think.